pykafka/simpleconsumer.py

```diff
@@ -408,37 +410,50 @@ def _handle_success(parts):
             to_retry.extend(parts_by_error.get(NotCoordinatorForConsumer.ERROR_CODE, []))
             reqs = [p.build_offset_fetch_request() for p, _ in to_retry]
 
-    def _reset_offsets(self, partitions=None):
-        """Reset offsets after an error
+    def reset_offsets(self, partitions=None):
+        """Reset offsets for the specified partitions
 
         Issue an OffsetRequest for each partition and set the appropriate
-        returned offset in the OwnedPartition per self._auto_offset_reset
+        returned offset in the OwnedPartition
 
-        :param partitions: the partitions for which to reset offsets
+        :param partitions: (`partition`, `offset`) pairs to reset
+            where `partition` is the partition for which to reset the offset
+            and `offset` is the new offset the partition should have
         :type partitions: Iterable of
-            :class:`pykafka.simpleconsumer.OwnedPartition`
+            (:class:`pykafka.simpleconsumer.OwnedPartition`, int)
```

Comment:
Oh, sure - if you're saving partition IDs to disk then you would have to do a bit of work to construct the input to `reset_offsets`. I think a cleaner solution is to stop this function from accepting `OwnedPartition` and have it take `Partition` instead, since `OwnedPartition` is supposed to be private. I'll hold off on merging until I've figured that out, then.